### PR TITLE
Fix condition for training on cpu

### PR DESCRIPTION
**Pull request by AlexanderDokuchaev**
For unknown reasons torch.cuda.is_available()' return false, when should be true.

### CODE REVIEW

**Comment by Ilya-Krylov** [comment]
@AlexanderDokuchaev I propose to retarget this PR to feature/ik/more_tools_with_update_config 
Also could you create a PR with this change (mmdetection commit update) to OTE (feature/mmedetection2.0) in order to see that tests are passed again.

**Comment by AlexanderDokuchaev** [comment]
run tests

File: mmdet/apis/train.py [hunk 1]
```diff
@@ -79,7 +80,10 @@ def train_detector(model,
     ]
 
     map_location = 'default'
-    if torch.cuda.is_available():
+    if os.getenv('CPU_ONLY', '0') == '1':
```
Comment (druzhkov-paul) [inline mmdet/apis/train.py:83]: Is it better than
```python 
if torch.cuda.is_available() and torch.cuda.device_count() > 0:
```
?

**Comment by druzhkov-paul** [comment]
> For unknown reasons torch.cuda.is_available()' return false, when should be true.

It may be because of mismatching CUDA version.

**Comment by Ilya-Krylov** [comment]
@druzhkov-paul do you know how determine this mismatching? For me torch.cuda.is_available() sometimes returns true and sometimes false. CI is also red, so that means that torch.cuda.is_avaliable() worked wrong by some reason.

**Comment by druzhkov-paul** [comment]
`torch.version.cuda` should show which CUDA version PyTorch is built with.
`torch.utils.cpp_extension.CUDA_HOME` should point to current CUDA runtime, I guess.
If PyTorch build version is higher than the one for runtime, then `torch.cuda.is_available()` will likely return `False`.

Also `tools/collect_env.py` script is useful for debugging of such issues.

**Comment by druzhkov-paul** [comment]
What I see from the logs, is that CUDA 10.1 runtime is used, but PyTorch build uses CUDA 10.2.

Target CUDA version for PyTorch can be explicitly specified as a part of a pip package name:
```
pip install torch==1.5.1+cu101 torchvision==0.6.1+cu101 -f https://download.pytorch.org/whl/torch_stable.html
```

**Comment by Ilya-Krylov** [comment]
@druzhkov-paul if verisions equality helps we have to fix CUDA vesrion in both CUDA requirenments and packages requirenments. Are you ok with that?

**Comment by druzhkov-paul** [comment]
Not sure that I understand the proposal.
We should definitely fix CUDA version in Docker files, but as for general requirements not sure that it'll be straightforward. BTW, we have a note about this in the [docs](https://github.com/opencv/mmdetection/blob/develop/docs/install.md#install-otedetection).

**Comment by AlexanderDokuchaev** [comment]
run tests

**Comment by Ilya-Krylov** [comment]
Had a discussion with @AlexanderDokuchaev , decided to close

**Pull request closed without merging**